src/app/shared/entry.ts

```diff
@@ -412,11 +412,14 @@ export abstract class Entry<V extends WorkflowVersion | Tag> implements OnDestro
    * Will decode the URL
    * @return {void}
    */
-  decodeURL(type: string): void {
-    const url = decodeURIComponent(window.location.href);
-    const containersIndex = this.getIndexInURL('/' + type);
-    const newPath = url.substring(containersIndex);
-    this.location.replaceState(newPath);
+  decodeURL(): void {
+    let url = decodeURIComponent(window.location.href);
+    // Get decoded path by extracting substring of url, starting from third forward slash '/'
+    // This removes the HTTP/HTTPS domain name (e.g. https://dockstore.org) from the url
+    for (let i = 0; i < 3; i++) {
+      url = i < 2 ? url.substring(url.indexOf('/') + 1) : url.substring(url.indexOf('/'));
+    }
+    this.location.replaceState(url);
```

Comment:
I think this change resolves some of the weird double domain URL paths. In the case of https://dockstore.org/services/asdf, the `type` passed to `decodeURL` is "workflows", which isn't present in the original URL and thus messes up the new URL at `replaceState`.

Seems to help keep the original URL for 404 paths that look like: `services/*`, `notebooks/*`, `apptools/*`, `search-workflows/*`.